insalan/payment/views.py

```diff
@@ -0,0 +1,160 @@
+
+        transaction_obj.payment_status = TransactionStatus.SUCCEEDED
+        transaction_obj.touch()
+        transaction_obj.save()
+
+        # Execute hooks
+        transaction_obj.run_success_hooks()
+
+        return Response(status=status.HTTP_200_OK)
+
+class ErrorView(generics.ListAPIView):
+    pass
+
+
+class PayView(generics.CreateAPIView):
+    permission_classes = [permissions.IsAuthenticated]
+    authentication_classes = [SessionAuthentication]
+    queryset = Transaction.objects.all()
+    serializer_class = serializers.TransactionSerializer
+
+    def create(self, request):
+        token = Token()
+        payer = request.user
+        data = request.data.copy()
+        data["payer"] = payer.id
+        logger.debug(f"data in view = {data}")  # contient des données
+        transaction = serializers.TransactionSerializer(data=data)
+        transaction.is_valid()
+        logger.debug(transaction.validated_data)
+        if transaction.is_valid(raise_exception=True):
+            transaction_obj = transaction.save()
+            # helloasso intent
+            helloasso_amount = int(
+                transaction_obj.amount * 100
+            )  # helloasso reads prices in cents
+            HELLOASSO_URL = getenv("HELLOASSO_ENDPOINT")
+            intent_body = {
+                "totalAmount": helloasso_amount,
+                "initialAmount": helloasso_amount,
+                "itemName": str(transaction_obj.id),
+                "backUrl": f"{getenv('HELLOASSO_BACK_URL')}?id={transaction_obj.id}",
+                "errorUrl": f"{getenv('HELLOASSO_ERROR_URL')}?id={transaction_obj.id}",
+                "returnUrl": f"{getenv('HELLOASSO_RETURN_URL')}?id={transaction_obj.id}",
+                "containsDonation": False,
+                "payer": {
+                    "firstName": payer.first_name,
+                    "lastName": payer.last_name,
+                    "email": payer.email,
+                },
+            }
+            headers = {
+                "authorization": "Bearer " + token.get_token(),
+                "Content-Type": "application/json",
+            }
+
+            checkout_init = requests.post(
+                f"{HELLOASSO_URL}/v5/organizations/insalan-test/checkout-intents",
+                data=json.dumps(intent_body),
+                headers=headers,
+                timeout=1
+            )  # initiate a helloasso intent
+            logger.debug(checkout_init.text)
+            checkout_json = checkout_init.json()
+            redirect_url = checkout_json["redirectUrl"]
+            intent_id = checkout_json["id"]
+            transaction_obj.intent_id = intent_id
+            transaction_obj.save()
+            logger.debug(intent_body)
+
+            # Execute hooks
+            transaction_obj.run_prepare_hooks()
+
+            return HttpResponseRedirect(redirect_to=redirect_url)
+        return JsonResponse({"problem": "oui"})
```

Comment:
Even though reading that made me burst out of laughter, a more explicit message would be more useful

Reply:
xD I didn't saw that, I was losing my mind at this point